src/category_theory/sparse.lean

```diff
@@ -0,0 +1,12 @@
+import category_theory.category
+
+universes u v
+
+namespace category_theory
+
+variables {C : Sort u} [category_struct.{v} C]
+
+-- This doesn't work well as an instance; use it to construct specific cases.
+def sparse_category [∀ X Y : C, subsingleton (X ⟶ Y)] : category.{v} C := { }
```

Comment:
This actually has an official name: *thin* or *posetal*: https://en.wikipedia.org/wiki/Posetal_category.

It is a little different than what I did in the Lean 2 HoTT library (it doesn't take care of identity morphisms, but it seems to work very nicely!